internal/graph/check.go

```diff
@@ -314,44 +336,64 @@ func exclusion(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHa
 		wg.Done()
 	}()
 
+	var dbReads uint32
 	for i := 0; i < len(handlers); i++ {
 		select {
 		case baseResult := <-baseChan:
 			if baseResult.err != nil {
-				return &openfgapb.CheckResponse{Allowed: false}, baseResult.err
+				return &ResolveCheckResponse{
+					Allowed:            false,
+					ResolutionMetadata: &ResolutionMetadata{},
```

Comment:
Do we have inconsistent behaviour?  In the union and intersection case, we will still return the dbRead that has been accumulated while we return 0 here.

Reply:
This is an error case, so if you have `define diff as a but not b` and  the evaluation of `a` fails, there is no point in tracking the `dbReads` 🤔

Reply:
Since we have no control over whether base/sub arrives first.  It is entirely be possible that we already made a db request for the sub.

In this case, it will analogous to union/intersection where one of the branch fail.  The logic for these cases are that we will return the number of db read

Reply:
Updated 👍